includes/transformer/class-post.php

```diff
@@ -143,15 +143,96 @@ protected function get_attributed_to() {
 	}
 
 	/**
-	 * Returns the Image Attachments for this Post, parsed from blocks.
+	 * Generates all Media Attachments for a Post.
+	 *
+	 * @return array The Attachments.
+	 */
+	protected function get_attachments() {
+		$max_images = intval( \apply_filters( 'activitypub_max_image_attachments', \get_option( 'activitypub_max_image_attachments', ACTIVITYPUB_MAX_IMAGE_ATTACHMENTS ) ) );
+
+		if ( site_supports_blocks() && \has_blocks( $this->wp_post->post_content ) ) {
+			return $this->get_block_attachments( $max_images );
+		}
+
+		return $this->get_classic_editor_images( $max_images );
+	}
+
+	/**
+	 * Get media attachments from blocks. They will be formatted as ActivityPub attachments, not as WP attachments.
+	 *
+	 * @param int $max_images The maximum number of attachments to return.
+	 *
+	 * @return array The attachments.
+	 */
+	protected function get_block_attachments( $max_images ) {
+		// max images can't be negative or zero
+		if ( $max_images <= 0 ) {
+			return array();
+		}
+
+		$id = $this->wp_post->ID;
+
+		$image_ids = array();
+
+		// list post thumbnail first if this post has one
+		if ( \function_exists( 'has_post_thumbnail' ) && \has_post_thumbnail( $id ) ) {
+			$image_ids[] = \get_post_thumbnail_id( $id );
```

Comment:
Note that I fixed the previous bug in the parser that led to #500 - it was because I did `$max_images--` here when that was incorrect for the parser, which needed the global value, including the thumbnail

Reply:
shouldn't we use `media_ids` or `attachment_ids` where we can?

I thinks we have to keep the `activitypub_max_image_attachments` for now, but that should be the only thing!?!

Reply:
Yup I think so, this was just a lot of copy-paste from the previous code as I restructured it.

I'll go and change the copy in the settings page while I'm at it to clarify that it will constrain audio/video as well